code/unit_tests/clothing.dm

```diff
@@ -0,0 +1,72 @@
+/datum/unit_test/gear_rig_validation_test
+	name = "GEAR: All Rigs Shall Have Non-Null Component Types"
+
+/datum/unit_test/gear_rig_validation_test/start_test()
+	var/list/results = list()
+	for(var/rigtype in subtypesof(/obj/item/rig))
+		var/res = ""
+		var/obj/item/rig/rig = new rigtype
+		if(initial(rig.gloves) == null)
+			res = "[res] - null initial gloves"
+		if(initial(rig.chest) == null)
+			res = "[res] - null initial chest"
+		if(initial(rig.helmet) == null)
+			res = "[res] - null initial helmet"
+		if(initial(rig.boots) == null)
+			res = "[res] - null initial boots"
```

Comment:
A minor issue as such but if there are multiple null components then you'll only get the last one (so would have to run the test multiple times to catch them all)

Reply:
Nevermind me, the `res` is "recursive"

Reply:
Yeah it's a bit unclear.